packages/frontend/src/hooks/useDownloadCsv.tsx

```diff
@@ -38,6 +41,40 @@ export const downloadCsv = async ({
     });
 };
 
+export const getCsvFileUrl = async (
```

Comment:
Nice! This makes the promise chain really tidy when using `pollCsvFileUrl` in components. Although we could let react-query handle the retries for us. That way the components don't even need to know about polling, they can just update on state changes from react query. 

I think the react query config is to have:

```
// just pseudo-code
{
  enabled: (response) => response.isCompleted,
  refetchInterval: 1000, #ms
}
```

Not sure which you think is cleaner.

Reply:
I was not sure how to build this with react-query, or its side effects... I feel in a case like this I prefer to be more explicit and have more control on how the code runs.